### ISSUE TITLE
Displaying package size before starting to download

### ISSUE THREAD

**Issue by salarmehr** [chocolatey/choco#701]
When some package are going to be installed/update via yum it first shows the size of each package and the sum of all packages that are going to be downloaded. This is very handy. May you please add similar behavior to choco?

Thanks.

**Comment by ferventcoder** [comment]
Have thought about it - however consider that you have two main types of packages with Chocolatey:

1. those that are truly packages and contain all of the software they represent in them - easy to show the size because the package size is the size
1. Packages that download the software they represent from official distribution points - hard to represent the full impact.

**Comment by ferventcoder** [comment]
And due to software redistribution laws, the publicly available packages on Chocolatey.org fall into the second category more often.

**Comment by ferventcoder** [comment]
We could definitely do this, but it would be further down the road.

**Comment by pyhedgehog** [comment]
We can have metadata of what will be downloaded. This can be checked by package-verifier. Look how it's implemented in gentoo portage.

**Comment by ferventcoder** [comment]
@pyhedgehog been thinking about how verifier and other things can pass back more information. This however would still be local only to the community repository though.

**Comment by tik9** [comment]
@ferventcoder first of thanks for all your good work. 

> Packages that download the software they represent from official distribution points.

 Is this software from the community which is difficult to size? How does Debian do it?

> local only to the community repository
What is this repo? Is it
> Community Package Repository Notification. There are 7936 Community Maintained Packages
Well, there is choco, Chrome, etc.

**Comment by YakovL** [comment]
For the second category, at least we could have a way for the distributor to set some metadata and choco could show something like "size (not verified): 123 kb"

**Comment by pauby** [comment]
Thanks for raising this issue. After reviewing it, it's not something we expect to prioritise in the foreseeable future. We’re closing it to keep our backlog focused, but we’re open to revisiting it in the future if priorities shift or new interest arises. You are welcome to comment here or open a new issue if you'd like to discuss it further down the line.
